packages/core/src/providers/netsuite.ts

```diff
@@ -0,0 +1,248 @@
+ *         userinfo: "https://1234567.restlets.api.netsuite.com/app/site/hosting/restlet.nl?script=123&deploy=1",
+ *         // Optional
+ *         prompt: "login", // Required if you want to force the user to login every time.
+ *         scope: "restlets", // Optional defaults to "restlets rest_webservices". Enter the scope(s) you want to use followed by spaces.
+ *       })
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *
+ * - [NetSuite - Creating an Integration Record (OAuth 2.0)](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771733782.html#Related-Topics)
+ * - [NetSuite - Authorizing OAuth Requests](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps)
+ * - [NetSuite - Configure OAuth Roles](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771510070.html#Set-Up-OAuth-2.0-Roles)
+ * - [Learn more about NetSuite OAuth 2.0](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/chapter_157769826287.html#OAuth-2.0)
+ * - [Source code](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/netsuite.ts)
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the NetSuite provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ * Make sure the `userinfo` matches the return type of the profile callback to ensure the user session gets read correctly.
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/providers/custom-provider#override-default-options).
+ *
+ * :::
+ *
+ * :::info **Disclaimer**
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function NetSuite<P extends NetSuiteProfile>(
+  config: OAuthNetSuiteOptions<P>
+): OAuthConfig<P> {
+  const { issuer, userinfo: userInfo, prompt = 'none', scope } = config
+
+  return {
+    id: "netsuite",
+    name: "NetSuite",
+    type: "oauth",
+    checks: ["state"],
+    authorization: {
+      url: `https://${issuer}.app.netsuite.com/app/login/oauth2/authorize.nl`,
+      params: {
+        client_id: `${config.clientId}`,
+        prompt: prompt,
+        response_type: 'code',
+        scope: scope || 'restlets rest_webservices',
+      }
+    },
+    token: {
+      url: `https://${issuer}.suitetalk.api.netsuite.com/services/rest/auth/oauth2/v1/token`,
+      params: {
+        grant_type: "authorization_code",
+      }
+    },
+    userinfo: `${userInfo}`,

```

Comment:
Is there a reason you're putting this string in a template literal? 

We coudl avoid renaming it to `userInfo` and just have this line be `userinfo`

```suggestion
    userinfo,
```

Reply:
Yes @ndom91 it's more so preference for readability. The template literal can be removed. There is an alias at the top where the destructuring is placed `userinfo: userInfo`. Having all lowercase for just that key and the rest aren't. It keeps the consistency and if it later gets changed from a lowercase key to a camelcase one it will be easy to refactor ensuring backwards compatibility.

Reply:
Right my point was just, if we skip the alias all together we can simplify this a bit (even though its then no longer camelCase haha)

Reply:
Either way, not a big deal. Let's try to get this in. 

Is there a public site i can make an account at that's using netsuite to test this?

Reply:
Haha I totally understand that 😄. There isn't a public site to test no. NetSuite is all blocked behind a paywall. But if you have the repo set up I can give you a test set of data? I do also have a working version of this used in production on a nextjs project. I can send you that as well? Whichever path works best for you. Lmk 😎

Reply:
@ndom91 let me know if you need assistance. I can test this with you if need be. Let me know!
![IGotYouThumbsUpGIF](https://github.com/nextauthjs/next-auth/assets/39228141/b62c2c79-e3e9-4ece-ad5c-4473687f7ae6)

Reply:
Hmm okay thats what I feared, that it was very enterprisey and not available to test for us.

Okay not a huge deal, we can merge it and deal with things as they come in. I assume NetSuite isn't going to be a hugely popular oauth provider :sweat_smile: